src/odemis/dataio/hdf5.py

```diff
@@ -1663,3 +1647,14 @@ def read_thumbnail(filename):
     # TODO: support filename to be a File or Stream
 
     return _thumbFromHDF5(filename)
+
+
+def read_str(s):
+    """
+    Make sure we can read both bytes (HDF5 ascii) and string (HDF5 utf8) metadata
+    That's important because sometimes strings are stored as bytes (ie ascii).
+    This forces both ascii and utf8 to be stored as strings (unicode)
```

Comment:
Reading this makes me quite confused. I'd suggest to clarify in this way:
```suggestion
    HDF5 has different types of strings: ASCII vs UTF8, and fixed- vs variable-length.
    h5py convert some to "unicode" (aka string) and some to bytes. Which one is converted
    to what depends on the version of h5py! To keep some sanity, when it is known that a
    data is a unicode text, use this function to ensure it.
```